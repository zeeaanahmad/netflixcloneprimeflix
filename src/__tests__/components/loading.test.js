import React from "react";
import { render } from "@testing-library/react";
import { Loading } from "../../components";

describe("<Loading/>", () => {
  it("renders the <Loading/> component", () => {
    const { container, getByTestId } = render(
      <Loading src="/images/test.png" data-testid="loading" />
    );

    expect(getByTestId("loading")).toBeTruthy();
    expect(getByTestId("loading-picture")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  describe("<Loading/>", () => {
    it("renders the <Loading.ReleaseBody/>", () => {
      const { container } = render(<Loading.ReleaseBody />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
