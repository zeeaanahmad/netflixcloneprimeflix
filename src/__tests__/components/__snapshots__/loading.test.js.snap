// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading/> <Loading/> renders the <Loading.ReleaseBody/> 1`] = `null`;

exports[`<Loading/> renders the <Loading/> component 1`] = `
<div
  class="loading__Spinner-sc-1qbyrzb-0 eyNPqo"
  data-testid="loading"
>
  <img
    class="loading__Picture-sc-1qbyrzb-1 ivSMRx"
    data-testid="loading-picture"
    src="/images/users//images/test.png.png"
  />
</div>
`;
